### ISSUE TITLE
Failed to restore: ftell: Illegal seek

### ISSUE THREAD

**Issue by avagin** [checkpoint-restore/criu#311]
```
[root@fc24 criu]# python test/zdtm.py run -t zdtm/static/env00 --remote -f ns
=== Run 1/1 ================ zdtm/static/env00

========================= Run zdtm/static/env00 in ns ==========================
Start test
./env00 --pidfile=env00.pid --outfile=env00.out --envname=ENV_00_TEST
Adding image cache
Adding image proxy
Run criu dump
Run criu restore
=[log]=> dump/zdtm/static/env00/31/1/restore.log
------------------------ grep Error ------------------------
RTNETLINK answers: File exists
(00.229895)      1: do_open_remote_image RDONLY path=route-9.img snapshot_id=dump/zdtm/static/env00/31/1
(00.230316)      1: 	Running ip route restore
Failed to restore: ftell: Illegal seek
(00.232757)      1: Error (criu/util.c:712): exited, status=255
(00.232777)      1: Error (criu/net.c:1479): IP tool failed on route restore
(00.232803)      1: Error (criu/net.c:2153): Can't create net_ns
(00.255091) Error (criu/cr-restore.c:1177): 105 killed by signal 9: Killed
(00.255307) Error (criu/mount.c:2960): mnt: Can't remove the directory /tmp/.criu.mntns.dTd7ak: No such file or directory
(00.255339) Error (criu/cr-restore.c:2119): Restoring FAILED.
------------------------ ERROR OVER ------------------------
################# Test zdtm/static/env00 FAIL at CRIU restore ##################
##################################### FAIL #####################################
```
@rodrigo-bruno @rppt

**Comment by rodrigo-bruno** [comment]
Hi,

I was trying to reproduce this in criu-dev but I got this error:

```
rbruno@metis ~/g/criu-dev> sudo test/zdtm.py run -t zdtm/static/env00  -f ns
=== Run 1/1 ================ zdtm/static/env00

========================= Run zdtm/static/env00 in ns ==========================
Start test
./env00 --pidfile=env00.pid --outfile=env00.out --envname=ENV_00_TEST
Run criu dump
=[log]=> dump/zdtm/static/env00/37/1/dump.log
------------------------ grep Error ------------------------
(00.003263) Dirty track supported on kernel
(00.003396) Found task size of 7ffffffff000
(00.005962) Warn  (criu/net.c:2497): Unable to get a socket network namespace
(00.005989) Warn  (criu/libnetlink.c:55): ERROR -95 reported by netlink
(00.005992) Error (criu/kerndat.c:572): NSID is not supported
(00.006018) Found mmap_min_addr 0x10000
(00.006026) Unlock network
(00.006029) Unfreezing tasks into 1
(00.006031) 	Unseizing 37 into 1
(00.006039) Error (compel/src/lib/infect.c:338): Unable to detach from 37: No such process
(00.006049) Error (criu/cr-dump.c:1701): Dumping FAILED.
------------------------ ERROR OVER ------------------------
################### Test zdtm/static/env00 FAIL at CRIU dump ###################
Send the 9 signal to  37
Wait for zdtm/static/env00(37) to die for 0.100000
##################################### FAIL #####################################
```

Do you have any idea why this happens?
I'm running Linux 3.19.0-64-generic and criu check does not complain.

In addition, I am able to sucessfully run this test using the remote option using the criu fork that I used to build the initial proxy/cache patch.

**Comment by rppt** [comment]
@avagin, I've also seen the "ftell: illegal seek" error.  My workaround was to make route*img O_FORCE_LOCAL to avoid redirection of stdin/out to/from socket. No idea what's the proper fix should be yet.

**Comment by xemul** [comment]
Can ip root restore routes/ifaddrs from stdin?

**Comment by rppt** [comment]
Yep:
https://git.kernel.org/pub/scm/linux/kernel/git/shemminger/iproute2.git/tree/ip/iproute.c#n1877

**Comment by xemul** [comment]
@rppt that's error check for EEXISTS from kernel RTNL req :)
Looks like it cannot:
https://git.kernel.org/pub/scm/linux/kernel/git/shemminger/iproute2.git/tree/ip/iproute.c#n1908

**Comment by rppt** [comment]
from man ip-restore:

```
       ip route restore
              restore routing table information from stdin
```

**Comment by xemul** [comment]
Sure, but try

    ip route restore < x.txt

versus

    cat x.txt | ip route restore

both are stdin-s but ftell() in the latter case would (should) fail.

**Comment by rst0git** [comment]
Resolved in checkpoint-restore@1d811bf2
